### ISSUE TITLE
[function-call-argument-newline] doesn't handle newlines inside arguments properly

### ISSUE THREAD

**Issue by fabiospampinato** [eslint/eslint#12419]
**Tell us about your environment**

* **ESLint Version:** 6.5.1
* **Node Version:** 10.15.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**

`@typescript-eslint/parser`

**Please show your full configuration:**

[Here](https://pastebin.com/Ni7HyKHL) it is, just uncomment the "function-call-argument-newline" rule.

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
fn ( 123, () => {
});

fn ( 123, () => {
}, [foo]);
```

```bash
npx eslint src --ext '.ts, .tsx'
```

**What did you expect to happen?**

No error detected.

**What actually happened? Please include the actual, raw output from ESLint.**

I think ESLint is kind of considering a newline _inside_ an argument (inside the body of the passed arrow function in this case) as a newline _between_ arguments, which is what should trigger this rule IMHO.

As a result of this misbehavior the first call to `fn` is fine but the second one is not.

Also this is a fairly common pattern to write when passing a dependency array to a React hook, so this would probably affect a lot of people.

**Are you willing to submit a pull request to fix this bug?**

I'm not too familiar with the AST, but with some pointers perhaps I can help.

**Comment by mdjermanovic** [comment]
This might be a duplicate of #12123

**Comment by fabiospampinato** [comment]
I'm closing this then.
